backend/dal/PIMSContext.cs

```diff
@@ -103,6 +109,31 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
             base.OnModelCreating(modelBuilder);
         }
 
+        public override int SaveChanges()
+        {
+            // get entries that are being Added or Updated
+            var modifiedEntries = ChangeTracker.Entries()
+                    .Where(x => (x.State == EntityState.Added || x.State == EntityState.Modified));
+            var userId = _httpContextAccessor.HttpContext.User.FindFirst(ClaimTypes.NameIdentifier).Value;
```

Comment:
Do you have access to the extension method `User.GetUserId()`?

Reply:
oh, right. let me see.